torch/utils/data/graph.py

```diff
@@ -0,0 +1,44 @@
+import io
+import pickle
+
+from torch.utils.data import IterableDataset
+
+from typing import Any, Dict
+
+reduce_ex_hook = None
+
+
+def stub_unpickler():
+    return "STUB"
+
+
+def list_connected_datapipes(scan_obj):
+
+    f = io.BytesIO()
+    p = pickle.Pickler(f)  # Not going to work for lambdas, but dill infinite loops on typing and can't be used as is
+
+    def stub_pickler(obj):
+        return stub_unpickler, ()
+
+    captured_connections = []
+
+    def reduce_hook(obj):
+        if obj == scan_obj:
+            raise NotImplementedError
+        else:
+            captured_connections.append(obj)
+            return stub_unpickler, ()
+
+    # TODO(VitalyFedyunin):  Better do it as `with` context for safety
```

Comment:
Agree with comment. We need to also take care of the existing `reduce_ex_hook` before we set hook to capture the connected datapipes.